torchdata/datapipes/iter/util/renamekeys.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import re
+from fnmatch import fnmatch
+from typing import Dict, Iterator, List, Union, Any
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+
+@functional_datapipe("rename_keys")
+class KeyRenamerIterDataPipe(IterDataPipe[Dict]):
+    r"""
+    Given a stream of dictionaries, rename keys using glob patterns.
+
+    This is used for quickly extracting relevant fields from a stream of dictionaries
+    and renaming them to a common format.
+
+    Note that if keys contain slashes, only the part after the last slash is matched.
+
+    Args:
+        source_datapipe: a DataPipe yielding a stream of dictionaries.
+        keep_unselected: keep keys/value pairs even if they don't match any pattern (False)
+        must_match: all key value pairs must match (True)
+        duplicate_is_error: it is an error if two renamings yield the same key (True)
+        *args: `(renamed, pattern)` pairs
+        **kw: `renamed=pattern` pairs
+
+    Returns:
+        a DataPipe yielding a stream of dictionaries.
+
+    Examples:
+        >>> dp = IterableWrapper([{"/a/b.jpg": b"data"}]).rename_keys(("image", "*.jpg"))
+        >>> list(dp)
+        [{'image': b'data'}]
+        >>> dp = IterableWrapper([
+            {"/a/b.input.jpg": b"data1", "/a/b.target.jpg": b"data2"},
+            {"/a/b.input.png": b"data1", "/a/b.target.png": b"data2"},
+        ]).rename_keys(input="*.input.*", output="*.target.*")
+        >>> list(dp)
+        [{'input': b'data1', 'target': b'data2'}, {'input': b'data1', 'target': b'data2'}]
+    """
+
+    def __init__(
+        self,
+        source_datapipe: IterDataPipe[Dict[Any, Any]],
+        *args,
+        keep_unselected=False,
+        must_match=True,
+        duplicate_is_error=True,
```

Comment:
```suggestion
        allow_duplicate=False,
```
nit: might be a better name but feel free to ignore